### PR TITLE
IGNITE-12177: Java compute tasks for C++

### PR DESCRIPTION
**Pull request by isapego**
1. Implemented `Compute::ExecuteJavaTask` methods family that allow execution of java tasks by name.
2. Added tests.
3. Fixed logic of creation of `Compute` objects on the given `ClusterGroup`

### The Contribution Checklist
- [x] There is a single JIRA ticket related to the pull request. 
- [x] The web-link to the pull request is attached to the JIRA ticket.
- [x] The JIRA ticket has the _Patch Available_ state.
- [x] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [x] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [x] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [x] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -1,776 +1,764 @@
-<profile version="1.0">
```
Comment (ptupitsyn) [inline .idea/inspectionProfiles/Project_Default.xml:1]: Is this huge change intentional?

File: modules/platforms/cpp/binary/include/ignite/impl/interop/interop_memory.h [hunk 2]
```diff
@@ -48,7 +48,19 @@ namespace ignite
 
             /** Flag: acquired. */
             const int IGNITE_MEM_FLAG_ACQUIRED = 0x4;
-                
+
+            union BinaryFloatInt32
```
Comment (ptupitsyn) [inline modules/platforms/cpp/binary/include/ignite/impl/interop/interop_memory.h:52]: Missing comments.

File: modules/platforms/cpp/core-test/include/ignite/compute_types.h [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+namespace ignite_test
+{
+    /*
+     * Check if cluster node contain the attribute with name provided.
+     */
+    class HasAttrName : public ignite::IgnitePredicate<ignite::cluster::ClusterNode>
+    {
+    public:
+        HasAttrName(std::string name) :
+        name(name)
```
Comment (ptupitsyn) [inline modules/platforms/cpp/core-test/include/ignite/compute_types.h:36]: Missing indentation.

File: modules/platforms/cpp/core-test/include/ignite/compute_types.h [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+#include <stdint.h>
+#include <string>
+
+#include <ignite/ignite_predicate.h>
+#include <ignite/cluster/cluster_node.h>
+
+namespace ignite_test
+{
+    /*
+     * Check if cluster node contain the attribute with name provided.
```
Comment (ptupitsyn) [inline modules/platforms/cpp/core-test/include/ignite/compute_types.h:30]: `contains an attribute`

File: modules/platforms/cpp/core/include/ignite/impl/compute/compute_impl.h [hunk 5]
```diff
@@ -198,7 +208,214 @@ namespace ignite
...
+                 * @param writer Binary writer.
+                 * @return Task result of type @c R.
+                 *
+                 * @tparam R Type of task result.
+                 * @tparam A Type of task argument.
+                 */
+                template<typename A>
+                void WriteJavaTask(const std::string& taskName, const A* arg, binary::BinaryWriterImpl& writer) {
+                    writer.WriteString(taskName);
+                    // Keep binary flag
```
Comment (ptupitsyn) [inline modules/platforms/cpp/core/include/ignite/impl/compute/compute_impl.h:321]: Missing blank line before comment.

**Review by ptupitsyn — APPROVED** [review]
Looks good to me, please see a few minor comments.

File: modules/platforms/cpp/binary/include/ignite/impl/interop/interop_memory.h (hunk 2 above)
Comment (isapego) [reply modules/platforms/cpp/binary/include/ignite/impl/interop/interop_memory.h:52]: Fixed.

File: modules/platforms/cpp/core-test/include/ignite/compute_types.h (hunk 3 above)
Comment (isapego) [reply modules/platforms/cpp/core-test/include/ignite/compute_types.h:36]: Fixed.

**Pull request closed without merging**